src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/ContentService.cs

```diff
@@ -389,94 +385,59 @@ public class ContentService : IContentService
                         return ValidationActionResult<CommentViewModel>(ContentBlockNotFound);
                     }
 
-                    var newComment = _context.Comment.Add(new Comment
+                    var comment = new Comment
                     {
                         Id = new Guid(),
-                        IContentBlockId = contentBlockId,
-                        Name = comment.Name,
-                        State = EnumUtil.GetFromString<CommentState>(comment.State),
-                        Time = comment.Time,
-                        CommentText = comment.CommentText,
-                        ResolvedBy = comment.ResolvedBy,
-                        ResolvedOn = comment.ResolvedOn,
-                    });
-
+                        ContentBlockId = contentBlockId,
+                        Content = request.Content,
+                        Created = DateTime.UtcNow,
+                        CreatedById = _userService.GetUserId()
+                    };
+                    
+                    await _context.Comment.AddAsync(comment);
                     await _context.SaveChangesAsync();
-                    return _mapper.Map<CommentViewModel>(newComment.Entity);
+
+                    var added = await _context.Comment
```

Comment:
Is there not a "GetCommentById" method we can call here already?

Reply:
No but I've added one. It's only private though because it knows nothing about the Release to check that with.